napi-inl.h

```diff
@@ -1461,26 +1517,30 @@ inline void Error::ThrowAsJavaScriptException() const {
   HandleScope scope(_env);
   if (!IsEmpty()) {
     napi_status status = napi_throw(_env, Value());
-    if (status != napi_ok) throw Error::New(_env);
+    NAPI_THROW_IF_FAILED(_env, status);
   }
 }
 
+#ifdef NAPI_CPP_EXCEPTIONS
+
 inline const char* Error::what() const NAPI_NOEXCEPT {
   return Message().c_str();
 }
 
+#endif // NAPI_CPP_EXCEPTIONS
```

Comment:
It would work. But when `NAPI_CPP_EXCEPTIONS` is not set, the `Error` class does not inherit from `std::exception`. In that case, it didn't seem to make sense to have a `what()` method that is otherwise inconsistent with the method naming style, and just a duplicate of `Message()`.